File: hospital/lib/models/Appointment.dart
class Appointment {
  final String date;
  final String time;
  final String status;
  final String Id;
  final String doctorId;
  final String patientId;

  Appointment({
    required this.date,
    required this.time,
    required this.status,
    required this.doctorId,
    required this.patientId,
    required this.Id,
  });

  Map<String, dynamic> toJson() {
    return {
      'date': date,
      'time': time,
      'status': status,
      'doctorId': doctorId,
      'patientId': patientId,
      'Id': Id,
    };
  }
}

File: hospital/server/models/appointment.js
const mongoose = require("mongoose");

const AppointmentSchema = new mongoose.Schema(
  {
    date: { type: String, trim: true },
    time: { type: String, trim: true },
    status: { type: String, trim: true },
    Id: { type: String, trim: true },
    doctorId: { type: String, trim: true },
    patientId: { type: String, trim: true },
  },
  { timestamps: true }
);

const Appointment = mongoose.model("Appointment", AppointmentSchema);
module.exports = Appointment;
